pkg/api/graph/graph.go

```diff
@@ -158,7 +162,7 @@ func (g Graph) SyntheticNodes() []graph.Node {
 	sort.Sort(SortedNodeList(nodeList))
 	for _, node := range nodeList {
 		if potentiallySyntheticNode, ok := node.(ExistenceChecker); ok {
-			if potentiallySyntheticNode.Found() {
+			if !potentiallySyntheticNode.Found() {
```

Comment:
Sure, I was just wondering if the bug affected code outside the scope of this PR (i.e. does it warrant a separate PR in advance of this one).